cpp/include/raft/core/macro_utils.hpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2019-2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __RAFT_RT_MACRO_UTILS_H
+#define __RAFT_RT_MACRO_UTILS_H
+
+#pragma once
+
+/**
+ * Some macro magic to remove optional parentheses of a macro argument.
+ * See https://stackoverflow.com/a/62984543
+ */
+#define RAFT_DEPAREN(X)      RAFT_DEPAREN_H2(RAFT_DEPAREN_H1 X)
```

Comment:
Does this file need to be part of the public API? I notice these are being moved out of the `detail` namespace for `rng`. I recently introduced a file with important macros in `raft/core/detail/macros.hpp` and it would be nice to consolidate these for consistency.